tests/integration/test_unknown_column_dist_table_with_alias/test.py

```diff
@@ -0,0 +1,34 @@
+import pytest
+from helpers.cluster import ClickHouseCluster
+import logging
+
+cluster = ClickHouseCluster(__file__)
+node = cluster.add_instance("node", main_configs=["configs/clusters.xml"])
+
+
+@pytest.fixture(scope="module")
+def start_cluster():
+    try:
+        logging.info("Starting cluster...")
+        cluster.start()
+        logging.info("Cluster started")
+
+        yield cluster
+    finally:
+        cluster.shutdown()
+
+
+@pytest.mark.parametrize("prefer_localhost_replica", [0, 1])
+def test_distributed_table_with_alias(start_cluster, prefer_localhost_replica):
+    node.query(
+        """
+        DROP TABLE IF EXISTS local;
+        DROP TABLE IF EXISTS dist;
+        CREATE TABLE local(`dummy` UInt8) ENGINE = MergeTree ORDER BY tuple();
+        CREATE TABLE dist AS local ENGINE = Distributed(localhost_cluster, currentDatabase(), local);
+    """
+    )
+
+    node.query(f"SET prefer_localhost_replica = {prefer_localhost_replica};")
```

Comment:
The `node.query()` has `settings` parameter. I'd suggest putting `prefer_localhost_replica` there. I'm not sure if it's guaranteed that `node.query()` calls are bound to the same client session, so setting a setting via separate query always has effect. This way, it's much more clear that the setting will be effective

Reply:
done